docs/coding-style.md

```diff
@@ -39,6 +39,14 @@ int CallerTwo() => Leaf() + PublicMethod();
 int Leaf() =>  42;
 ```
 
+#### Local functions
+
+There are no strict rules on when to use local functions. It should be decided on a case-by-case basis.
+
+By default, you should prefer methods over local functions. Use local functions if it makes the code significantly easier to understand. For example:
+- Accessing the method's local state directly, instead of using parameters, reduces noise.
+- The name of the function would not make sense at the class level.
+
 Local functions should always be placed at the end of a method.
```

Comment:
While you're on it, you can fix the syntax error on line 57, `return` is not valid with arrow method